benches/src/main.rs

```diff
@@ -15,17 +22,91 @@ use honeycomb_benches::{
 fn main() {
     let cli = Cli::parse();
 
+    let n_t = if let Some(val) = cli.n_threads {
+        val.get()
+    } else {
+        std::thread::available_parallelism()
+            .map(|v| v.get())
+            .unwrap_or(1)
+    };
+    let builder = ThreadPoolBuilder::new().num_threads(n_t);
+
+    #[cfg(feature = "thread-binding")]
+    {
+        if cli.bind_threads {
+            // build the topology & check that all necessary features are available on the machine
+            let topology = Topology::new().unwrap();
+            let topology = Arc::new(topology);
+            match get_physical_cores(&topology) {
+                Ok(mut cores) => {
+                    let n_t = if cores.len() < n_t {
+                        // don't allow more than one thread per physical core
+                        // this is sane since this branch executes only if we explicitly enable binding
+                        eprintln!(
+                            "W: Less physical cores than logical threads; proceeding with one thread per core ({})",
+                            cores.len()
+                        );
+                        cores.len()
```

Comment:
I think it is a better idea to allow oversubscription. Hyper-threading or even just multiple concurrent threads can be beneficial to hide latency.